src/events.js

```diff
@@ -3660,76 +2501,151 @@ export const addEventFilter = () => {
         addPaginationFunctionality(numPages, filter);
+    });
+}
 
-    })
+/**
+ * Search for the specimenId in the shipped boxes
+ * @param {array<object>} allBoxesList - list of all boxes for the healthcare provider
+ * @param {string} masterSpecimenId - specimenId to search for
+ * @returns {boolean} true if the specimenId is found in the shipped boxes, false otherwise.
+ * If the specimenId is a masterId, search for the key in the bags.
+ * If the bag is unlabelled, search arrElements for the key.
+ * If the specimen isn't a masterId, iterate the bags and search for the keys in the arrElements. 
+ * TODO: future implementation will include shipped property on the specimen object
+ */
+export const isScannedIdShipped = (allBoxesList, masterSpecimenId) => {
+    if (!allBoxesList.length) return false;
 
-}
+    const specimenIdType = masterSpecimenId.split(' ')[1];
+    const isMasterId = ['0007', '0008', '0009'].includes(specimenIdType);
 
-export const isScannedIdShipped = (getAllBoxesWithoutConversionResponse, masterSpecimendId) => {
-  const getAllBoxesWithoutConversionData = getAllBoxesWithoutConversionResponse.data;
-  if(!getAllBoxesWithoutConversionData.length) return false;
-  const inputScanned = masterSpecimendId;
-  const allBoxesShippedFilter = getAllBoxesWithoutConversionData.filter(box => box.hasOwnProperty('145971562')) // Submit shipment flag - '145971562'
-  let foundMatch = false
+    for (const box of allBoxesList) {
+        if (box[conceptIds.submitShipmentFlag] !== conceptIds.yes) continue;
+        
+        const { bags } = box;
+        if (isMasterId) {
+            if (bags[masterSpecimenId] || (bags.unlabelled && bags.unlabelled.arrElements.includes(masterSpecimenId))) {
+                return true;
+            }
+        } else {
+            for (const bagId in bags) {
+                if (bags[bagId].arrElements.includes(masterSpecimenId)) {
+                    return true;
+                }
+            }
+        }
+    }
 
-  for (let box of allBoxesShippedFilter) {
-      if(foundMatch) break;
-      
-      for(let bagConceptId of bagConceptIdList) {
-        const bag = box[bagConceptId];
-        if (
-          bag?.["223999569"] == inputScanned || // Biohazard Bag (mouthwash) scan
-          bag?.["522094118"] == inputScanned || // Orphan Bag/Container Scan
-          bag?.["787237543"] == inputScanned || // Biohazard Bag (Blood or Blood/Urine) ID
-          bag?.["234868461"]?.includes(inputScanned)){ // Check if input is found in (Samples Within - "234868461") array
-          foundMatch = true;
-          break;
+    return false;
+};
+
+const findScannedIdInUnshippedBoxes = (allBoxesList, masterSpecimenId) => {
+    const buildFoundDataObj = (box) => {
+        const dataObj = {};
+        dataObj['foundMatch'] = true;
+        dataObj[conceptIds.shippingLocation] = box[conceptIds.shippingLocation];
+        dataObj[conceptIds.shippingBoxId] = box[conceptIds.shippingBoxId];
```

Comment:
```suggestion
        dataObj[conceptIds.shippingBoxId] = box[conceptIds.shippingBoxId];
        dataObj['inputScanned'] = masterSpecimenId;
```

The `inputScanned` key and `masterSpecimenId` value may need to be added to the `dataObj` so `inputScanned` can be referenced from the object in the `If(isScannedInUnshippedBoxes) {...}` [conditional](https://github.com/episphere/biospecimen/pull/568/files#diff-aa3209e98840e6d236562086e3c6942a28f0c6beef6897d036367d09cc313c59R204-R210).

Reply:
committed per your suggestion @Gbarra9